Source/WebCore/platform/mediastream/gstreamer/RealtimeOutgoingMediaSourceGStreamer.cpp

```diff
@@ -68,9 +68,11 @@ RealtimeOutgoingMediaSourceGStreamer::~RealtimeOutgoingMediaSourceGStreamer()
         gst_element_release_request_pad(parent.get(), m_webrtcSinkPad.get());
     }
 
+    gst_element_set_locked_state(m_bin.get(), TRUE);
     gst_element_set_state(m_bin.get(), GST_STATE_NULL);
     if (auto pipeline = adoptGRef(gst_element_get_parent(m_bin.get())))
         gst_bin_remove(GST_BIN_CAST(pipeline.get()), m_bin.get());
+    gst_element_set_locked_state(m_bin.get(), FALSE);
```

Comment:
It might be interesting to write an automatic locker for this one as well. No strong opinion.

Reply:
Maybe later!